cpp/cmake_modules/DefineOptions.cmake

```diff
@@ -387,6 +387,7 @@ takes precedence over ccache if a storage backend is configured" ON)
 (This is a deprecated option. Use CMake presets instead.)"
                 OFF
                 DEPENDS
+                ARROW_ACERO
```

Comment:
I am not entirely sure why are we doing this change. Should all ARROW_PYTHON dependencies be added? what about S3 or GCS?

Reply:
Also, shouldn't we in theory _not_ rely on this in our own CI, given that it is a deprecated option?

If the minimal python build needs this, then we should just enable ARROW_ACERO directly in that build? (or update it to not require acero)

Reply:
Sorry. This is needless.
I thought that this is the cause of #36680 but it was wrong: #36683

I thought that this is still needed because #34711 removes `ARROW_COMPUTE` from `ARROW_PYTHON` instead of replacing `ARROW_COMPUTE` with `ARROW_ACERO`.
But it's correct because `ARROW_PYTHON` depends on `ARROW_ACERO` explicitly: `ARROW_PYTHON` -> `ARROW_DATASET` -> `ARROW_ACERO`